demo/src/main/java/com/google/android/fhir/demo/data/TimestampBasedDownloadWorkManagerImpl.kt

```diff
@@ -96,7 +104,8 @@ class TimestampBasedDownloadWorkManagerImpl(
             val toBeRequestedIds = searchIds.take(resourceFetchByIdRequestSize).toSet()
             val leftOverIds = searchIds - toBeRequestedIds
             resourceReferences[resourceType]!![searchParameter] = leftOverIds
-            val request = createUrlSearchRequestFromIds(resourceType,searchParameter , toBeRequestedIds)
+            val request =
```

Comment:
Remove the key if leftOverIds.size is 0 ?
And then remove the parent key if no values ?

Reply:
Doesnt change any flow because empty ids list is being handled when creating Request object.